### PR TITLE
pass azure IOT secrets to composite actions via the inputs

### PR DESCRIPTION
**Pull request by yash-zededa**
With reference to the [PR-1018](https://github.com/lf-edge/eden/pull/1018).

Secrets defined in the `env` section of a GitHub Actions workflow cannot be accessed directly within composite actions. Instead, you need to pass the secrets as inputs to the composite action and then map them as environment variables within the action.

### CODE REVIEW

**Review by shjala — APPROVED** [review]
Is there any way to test this before merging?

**Comment by yash-zededa** [comment]
@shjalaI  don’t have merge access for the Eden repo. If you do, please feel free to merge the PRs.

**Comment by uncleDecart** [comment]
I will merge this once I check it...

**Comment by yash-zededa** [comment]
> Is there any way to test this before merging?

There is one way. We need to modify test.yml to be triggered on pull_request_target - this way GHA would be able to read the secrets and then ideally the secrets would be passed to the composite actions. But to test the end to end the changes of the [PR-1011](https://github.com/lf-edge/eden/pull/1011) should be in this PR.

**Review by uncleDecart — APPROVED** [review]

**Pull request merged**
